### PR TITLE
Aha/sirius update

### PR DESCRIPTION
**Pull request by andyHa**

### CODE REVIEW

File: src/main/java/sirius/biz/datapipes/memory/Allocator.java [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+ * http://www.scireum.de - info@scireum.de
+ */
+
+package sirius.biz.datapipes.memory;
+
+import sirius.kernel.nls.NLS;
+
+import java.util.concurrent.atomic.AtomicLong;
+
+public class Allocator {
```
Comment (tbiScireum) [inline src/main/java/sirius/biz/datapipes/memory/Allocator.java]: javadoc: what is the purpose of this class ? 
javadoc for all new classes ?! and and some of the methods ?

File: src/main/java/sirius/biz/isenguard/Isenguard.java [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+
+    private boolean isRateLimitReached(String ip, String realm, int intervalInSeconds, int limit) {
+        long currentInterval = (System.currentTimeMillis() / 1000) / intervalInSeconds;
+        String key = "rate-limit-" + ip + "-" + realm + "-" + currentInterval;
+        return redis.query(() -> "Rate Limit", db -> {
+            long value = db.incr(key);
+            if (value == 1) {
+                db.expire(key, intervalInSeconds);
+            } else if (value == limit) {
+                //TODO log
```
Comment (tbiScireum) [inline src/main/java/sirius/biz/isenguard/Isenguard.java]: Ticket number ?

File: src/main/java/sirius/biz/isenguard/Isenguard.java [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+            }
+            return value >= limit;
+        });
+    }
+}
+//
+//    -Multi-Redis
```
Comment (tbiScireum) [inline src/main/java/sirius/biz/isenguard/Isenguard.java]: ?

File: src/main/java/sirius/biz/datapipes/memory/LargeMemory.java [hunk 4]
```diff
@@ -0,0 +1,430 @@
...
+            UNSAFE.putLong(buffers[bufferIndex] + bufferOffset, data);
+        }
+    }
+
+    protected void writeLongCrossBoundary(long address, long data) {
+        if (BIG_ENDIAN) {
+            writeBytes(address,
+                       new byte[]{(byte) (data & 0xFFL),
+                                  (byte) ((data & 0xFF00L) >> 8),
+                                  (byte) ((data & 0xFF000L) >> 16),
```
Comment (jmuscireum) [inline src/main/java/sirius/biz/datapipes/memory/LargeMemory.java]: Should be FF0000 if I am correct

File: src/main/java/sirius/biz/datapipes/memory/LargeMemory.java [hunk 5]
```diff
@@ -0,0 +1,430 @@
...
+                                  (byte) ((data & 0xFF00000000000000L) >> 56)},
+                       4);
+        } else {
+            writeBytes(address,
+                       new byte[]{(byte) ((data & 0xFF00000000000000L) >> 56),
+                                  (byte) ((data & 0xFF000000000000L) >> 48),
+                                  (byte) ((data & 0xFF0000000000L) >> 40),
+                                  (byte) ((data & 0xFF00000000L) >> 32),
+                                  (byte) ((data & 0xFF000000) >> 24),
+                                  (byte) ((data & 0xFF000) >> 16),
```
Comment (jmuscireum) [inline src/main/java/sirius/biz/datapipes/memory/LargeMemory.java]: Should be FF0000 if I am correct

File: src/main/java/sirius/biz/datapipes/memory/LargeMemory.java [hunk 6]
```diff
@@ -0,0 +1,430 @@
...
+            writeBytes(address,
+                       new byte[]{(byte) (data & 0xFF),
+                                  (byte) ((data & 0xFF00) >> 8),
+                                  (byte) ((data & 0xFF000) >> 16),
+                                  (byte) ((data & 0xFF000000) >> 24)},
+                       4);
+        } else {
+            writeBytes(address,
+                       new byte[]{(byte) ((data & 0xFF000000) >> 24),
+                                  (byte) ((data & 0xFF000) >> 16),
```
Comment (jmuscireum) [inline src/main/java/sirius/biz/datapipes/memory/LargeMemory.java]: FF0000?

File: src/main/java/sirius/biz/datapipes/memory/LargeMemory.java [hunk 7]
```diff
@@ -0,0 +1,430 @@
...
+            UNSAFE.putInt(buffers[bufferIndex] + bufferOffset, data);
+        }
+    }
+
+    protected void writeIntCrossBoundary(long address, int data) {
+        if (BIG_ENDIAN) {
+            writeBytes(address,
+                       new byte[]{(byte) (data & 0xFF),
+                                  (byte) ((data & 0xFF00) >> 8),
+                                  (byte) ((data & 0xFF000) >> 16),
```
Comment (jmuscireum) [inline src/main/java/sirius/biz/datapipes/memory/LargeMemory.java]: FF0000?

File: src/main/java/sirius/biz/protocol/CleanupProtocolsTask.java [hunk 8]
```diff
@@ -28,25 +29,75 @@
...
 
     @Override
     public void runTimer() throws Exception {
+        if (elastic.getReadyFuture().isCompleted()) {
+            tasks.defaultExecutor().fork(() -> {
+                deleteLogs();
+                deleteIncidents();
+                deleteMails();
+                deleteJournal();
+                //TODO auditlogs
```
Comment (tbiScireum) [inline src/main/java/sirius/biz/protocol/CleanupProtocolsTask.java]: ticket number ?

File: src/main/java/sirius/biz/protocol/IncidentController.java [hunk 9]
```diff
@@ -0,0 +1,69 @@
...
+     */
+    @Permission(Protocols.PERMISSION_SYSTEM_PROTOCOLS)
+    @DefaultRoute
+    @Routed("/system/errors")
+    public void errors(WebContext ctx) {
+        ElasticPageHelper<StoredIncident> ph = ElasticPageHelper.withQuery(elastic.select(StoredIncident.class)
+                                                                                  .orderDesc(StoredIncident.LAST_OCCURRENCE));
+        ph.withContext(ctx);
+        ph.addTermAggregation(StoredIncident.CATEGORY);
+        ph.addTimeAggregation(StoredIncident.LAST_OCCURRENCE,
```
Comment (tbiScireum) [inline src/main/java/sirius/biz/protocol/IncidentController.java]: can we change the page size for this page to 100?

File: src/main/java/sirius/biz/reports/Reports.java [hunk 10]
```diff
@@ -0,0 +1,185 @@
...
+package sirius.biz.reports;
+
+import sirius.kernel.di.std.Register;
+
+/**
+ * Provides helper methods to create reports.
+ */
+@Register(classes = Reports.class)
+public class Reports {
+//
```
Comment (tbiScireum) [inline src/main/java/sirius/biz/reports/Reports.java]: ? delete or TODO ?

File: src/main/java/sirius/biz/reports/ReportsController.java [hunk 11]
```diff
@@ -0,0 +1,73 @@
...
+
+package sirius.biz.reports;
+
+import sirius.kernel.di.std.Register;
+import sirius.web.controller.BasicController;
+import sirius.web.controller.Controller;
+
+@Register(classes = Controller.class)
+public class ReportsController extends BasicController {
+//
```
Comment (tbiScireum) [inline src/main/java/sirius/biz/reports/ReportsController.java]: Delete or TODO ?

File: src/main/java/sirius/biz/s3/ObjectStore.java [hunk 12]
```diff
@@ -0,0 +1,69 @@
...
+import sirius.kernel.cache.CacheManager;
+import sirius.kernel.health.Exceptions;
+
+import java.time.LocalDate;
+
+public class ObjectStore {
+
+    protected final AmazonS3Client client;
+    protected final String bucketSuffix;
+    //TODO bad idea
```
Comment (tbiScireum) [inline src/main/java/sirius/biz/s3/ObjectStore.java]: Ticket ?

File: src/main/java/sirius/biz/web/TenantAware.java [hunk 13]
```diff
@@ -36,5 +36,6 @@ default void assertSameTenant(Supplier<String> fieldLabel, TenantAware other) {
         }
     }
 
+    //TODO
```
Comment (tbiScireum) [inline src/main/java/sirius/biz/web/TenantAware.java]: Ticket

File: src/main/java/sirius/biz/work/DistributedTasks.java [hunk 14]
```diff
@@ -0,0 +1,112 @@
...
+        }
+        return getSemaphore(concurrencyToken).tryAcquire();
+    }
+
+    private Semaphore getSemaphore(String name) {
+        return null ; //concurrencyTokens.computeIfAbsent(name, () -> new Semaphore());
+    }
+
+    private Runnable pullWork(DistributedTaskExecutor executor) {
+//        String task = redis.pollQueue(executor.getName());
```
Comment (tbiScireum) [inline src/main/java/sirius/biz/work/DistributedTasks.java]: TODO ?

File: src/main/java/sirius/biz/work/DistributedTasks.java [hunk 15]
```diff
@@ -0,0 +1,112 @@
...
+    private boolean aquireConcurrencyToken(DistributedTaskExecutor executor) {
+        String concurrencyToken = executor.getConcurrencyToken();
+        if (Strings.isEmpty(concurrencyToken)) {
+            return true;
+        }
+        return getSemaphore(concurrencyToken).tryAcquire();
+    }
+
+    private Semaphore getSemaphore(String name) {
+        return null ; //concurrencyTokens.computeIfAbsent(name, () -> new Semaphore());
```
Comment (tbiScireum) [inline src/main/java/sirius/biz/work/DistributedTasks.java]: TODO ?

File: src/main/java/sirius/biz/datapipes/memory/LargeMemory.java [hunk 16]
```diff
@@ -0,0 +1,430 @@
...
+
+import sirius.kernel.commons.Strings;
+import sirius.kernel.nls.NLS;
+import sun.misc.Unsafe;
+
+import java.lang.reflect.Field;
+import java.nio.ByteOrder;
+import java.util.concurrent.atomic.AtomicLong;
+
+public class LargeMemory {
```
Comment (tbiScireum) [inline src/main/java/sirius/biz/datapipes/memory/LargeMemory.java]: tests ?

File: src/main/resources/biz_de.properties [hunk 17]
```diff
@@ -1,257 +1,258 @@
...
-UserAccountController.noUserFoundForEmail = Für die angegebene eMail-Adresse wurde kein Benutzer gefunden.
-UserAccountController.tooManyUsersFoundForEmail = Die angegebene eMail-Adresse ist nicht eindeutig.
-VirtualObject.download = Herunterladen
-VirtualObject.fileSize = Größe
-VirtualObject.physicalKey = Physikalischer Schlüssel
-VirtualObject.rename = Umbenennen
-VirtualObject.replaceContents = Inhalt ersetzen
-VirtualObject.unreference = Referenzierung löschen
-VirtualObject.versions = Versionen
+Model.create=Neu Erstellen
```
Comment (jmuscireum) [inline src/main/resources/biz_de.properties]: These changes don't comply with our new Lokalise format

**Pull request merged**